src/RoleObjectFactory.js

```diff
@@ -365,6 +361,12 @@ function createAccessibilityObjectForRole(config) {
     } else {
       parent.accessible.addChild(displayObject);
     }
+  } else {
```

Comment:
The body of this `else` should be moved to the body of the deleted if-statement on what was lines 89-92. By updating those variables earlier, the right values can be extracted from the existing parent, and the if-statement on lines 362-367 will work with the new values.